### PR TITLE
Visual refresh replace message box

### PR DESCRIPTION
**Pull request by filipeotero**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

This PR is to replace the native MesssageBox with the visual refresh MessageBoxService.

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

@QilongTang 

### FYIs

@RobertGlobant20 @jesusalvino

### CODE REVIEW

**Comment by reddyashish** [comment]
@filipeotero Any instances of MessageBox to be changed in test files?

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -1792,7 +1792,7 @@ internal bool CanSaveAs(object parameters)
         private void InternalSaveAs(string path, SaveContext saveContext, bool isBackup = false)
         {
             try
-            {
+            {
```
Comment (zeusongit) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:1795]: extra spaces

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 2]
```diff
@@ -1620,7 +1620,7 @@ private void Open(object parameters)
                         errorMsgString = String.Format(Resources.MessageUnkownErrorOpeningFile, filePath);
                     }
                     model.Logger.LogNotification("Dynamo", commandString, errorMsgString, e.ToString());
-                    System.Windows.MessageBox.Show(errorMsgString);
+                    MessageBoxService.Show(errorMsgString, string.Empty, MessageBoxButton.OK, MessageBoxImage.None);
```
Comment (zeusongit) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:1623]: :+1:

**Review by zeusongit — APPROVED** [review]

**Comment by reddyashish** [comment]
This has to be changed as well https://github.com/DynamoDS/Dynamo/blob/2f9354254b83e9265d45a192a4e6e69c5d629eae/src/Tools/InstallUpdate/Program.cs#L146 ?

**Comment by filipeotero** [comment]
> @filipeotero Any instances of MessageBox to be changed in test files?

Not really. The MessageBoxes used in unit tests are the `MessageBoxService` 👍

**Comment by filipeotero** [comment]
> 

This part of the code is not accessible because `MessageBoxService` is an internal class. Should we make it public?

**Comment by filipeotero** [comment]
> 



> This has to be changed as well ?
> 
> https://github.com/DynamoDS/Dynamo/blob/2f9354254b83e9265d45a192a4e6e69c5d629eae/src/Tools/InstallUpdate/Program.cs#L146

This part of the code is not accessible because `MessageBoxService `is an internal class. Should we make it public?

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs (hunk 1 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:1795]: @filipeotero Can you address this as well before we merge?

**Pull request merged**